core/pkg/model/customcostresponse.go

```diff
@@ -2,60 +2,136 @@ package model
 
 import "github.com/opencost/opencost/core/pkg/opencost"
 
+// see design at https://link.excalidraw.com/l/ABLQ24dkKai/CBEQtjH6Mr
+// for additional details on how these objects work in the context of
+// opencost's plugin system
+
 type CustomCostResponse struct {
-	Metadata   map[string]string
+	// provides metadata on the Custom CostResponse
+	// deliberately left unstructured
+	Metadata map[string]string
+	// declared by plugin
+	// eg snowflake == "data management",
+	// datadog == "observability" etc
+	// intended for top level agg
 	Costsource string
-	Domain     string
-	Version    string
-	Currency   string
-	Window     opencost.Window
-	Costs      []*CustomCost
-	Errors     []error
-}
-
+	// the name of the custom cost source
+	// e.g., "datadog"
+	Domain string
+	// the version of the Custom Cost response
+	// is set by the plugin, will vary between
+	// different plugins
+	Version string
+	// FOCUS billing currency
+	Currency string
+	// the window of the returned objects
+	Window opencost.Window
+	// array of CustomCosts
+	Costs []*CustomCost
+	// any errors in processing
+	Errors []error
+}
+
+// designed to provide a superset of the FOCUS spec
+// https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/releases/latest/download/spec.pdf
 type CustomCost struct {
-	Metadata       map[string]string
-	Zone           string
-	BilledCost     float32
-	AccountName    string
+	// provides metadata on the Custom CostResponse
+	// deliberately left unstructured
+	Metadata map[string]string
+	// the region that the resource was incurred
+	// corresponds to 'availability zone' of FOCUS
+	Zone string
+	// FOCUS billed Cost
+	BilledCost float32
```

Comment:
focus doesn't specify anything other than 'numeric'. quote from the spec: 

> The FOCUS specification does not require a specific level of precision for numeric values. The level of precision required for a given column is determined by the provider and should be part of a data definition published by the provider.

i picked 32 bit since that seemed accurate enough, but open to a different type

Reply:
I guess this LGTM then; people always say that money numbers should be represented in a better type than floating-point https://stackoverflow.com/questions/15726535/which-datatype-should-be-used-for-currency, but I'm guessing it gets very complicated for non-US currencies. Still, it'd be nice to multiply by 100 and store in cents.

Reply:
For an async conversation (not blocking merge): @mbolt35 and @nikovacevic have you given any deeper thought to representations of currency?